web/genericl10n.js

```diff
@@ -15,63 +15,68 @@
 
 /** @typedef {import("./interfaces").IL10n} IL10n */
 
-import "../external/webL10n/l10n.js";
-import { getL10nFallback } from "./l10n_utils.js";
-
-const PARTIAL_LANG_CODES = {
-  en: "en-US",
-  es: "es-ES",
-  fy: "fy-NL",
-  ga: "ga-IE",
-  gu: "gu-IN",
-  hi: "hi-IN",
-  hy: "hy-AM",
-  nb: "nb-NO",
-  ne: "ne-NP",
-  nn: "nn-NO",
-  pa: "pa-IN",
-  pt: "pt-PT",
-  sv: "sv-SE",
-  zh: "zh-CN",
-};
-
-// Try to support "incompletely" specified language codes (see issue 13689).
-function fixupLangCode(langCode) {
-  return PARTIAL_LANG_CODES[langCode?.toLowerCase()] || langCode;
-}
+import { FluentBundle, FluentResource } from "fluent-bundle";
+import { DOMLocalization } from "fluent-dom";
+import { L10n } from "./l10n.js";
 
 /**
  * @implements {IL10n}
  */
-class GenericL10n {
+class GenericL10n extends L10n {
   constructor(lang) {
-    const { webL10n } = document;
-    this._lang = lang;
-    this._ready = new Promise((resolve, reject) => {
-      webL10n.setLanguage(fixupLangCode(lang), () => {
-        resolve(webL10n);
-      });
-    });
-  }
-
-  async getLanguage() {
-    const l10n = await this._ready;
-    return l10n.getLanguage();
+    super({ lang });
+    this.setL10n(
+      new DOMLocalization(
+        [],
+        GenericL10n.#generateBundles.bind(
+          GenericL10n,
+          "en-US",
+          this.getLanguage()
```

Comment:
Given that this method is `async`, how does this work in `#generateBundles` method since that it (according to the JSDocs) expects a string argument?